specs/time_series_api.adoc

```diff
@@ -0,0 +1,285 @@
+
+A small introduction to the components part of this spec
+(borrowed from Tendrl/documentation):
+
+=== Tendr/tendrl-api
+
+* Frontend/northbound API.
+* This is the integration point for any external management system/frontend
+  with Tendrl, including the Tendrl UI.
+* The App has the necessary logic to understand the state definitions stored in
+  the Central Store and present them over it’s API.
+* The App has the necessary logic to be able to invoke Operations through the
+  Central Store’s Job Queue implementation.
+* The App is completely stateless. All the state is stored in the Central
+  Store. The App essentially acts as a translation proxy between an external
+  system (including Tendrl UI) and the Central Store.
+* The App is intended to be primarily used for systems integration. As such,
+  the API is completely generic and self descriptive. The intended use case is
+  that any system wishing to leverage Tendrl uses the API’s descriptive
+  responses to construct targeted queries for a specific cluster instances
+  being managed by Tendrl.
+
+=== Tendrl Performance (performance_monitoring)
+
+It is a separate service that runs outside tendrl core stack and it relies on
+the state information in tendrl’s central store for its functioning. Typically
+it can be co-resident with tendrl application but nothing necessiates this.
+Following are its sub-components based on the functionalities:
+
+* Api Layer : This layer will serve the following responsibilities.
+  ** Serve the time series data stored in time series database.
+  ** The apis internally use a plugin based approach for querying the
+     time-series performance data which eases the task of supporting multiple
+     time-series databases.
+* Aggregator : This module aggregates atomic stats(ex: cluster utilization) to
+  composite(system utilization) stats which will be required by UI for the
+  dashboards.
+
+== Problem description
+
+Tendrl/tendrl-api(a part of tendrl-core) is agnostic of time-series db(a part
+of monitoring stack of tendrl) and its specifics but needs to be able to expose
+data in the time series db if the monitoring stack is installed.
+
+== Use Cases
+
+* If monitoring stack is installed by user, the user should be able to seamlessly
+  access tendrl maintained time series performance data over specified
+  time-ranges.
+
+== Proposed change
+
+* Tendrl/performance-monitoring will expose apis to access performance data in
+  time-series db. This is achieved by:
+  ** Interface to time-series db is designed in the form of plugins specific to
+     a time-series db.
+    *** This means that to support a new time-series db, all it takes is the
+        addition of its apecic interface as a plugin and doesn't require
+        touching any other part of the code.
+    *** The way the plugins work is :
+      **** The interfaces to time-series db are added under the directory
+           tendrl/performance-monitoring/time_series_db/dbplugins of
```

Comment:
If these are etcd directories, please use "_tendrl/" instead of "tendrl/", Etcd hides directories with a leading "_" and we need to use it